### PR TITLE
Possible fix for issue #241

### PR DESCRIPTION
**Pull request by cthare**
I'm a bit new to working on larger projects and open source in general. I think I may have solved #241.

### CODE REVIEW

File: extras/dummy_generator/generator.py [hunk 1]
```diff
@@ -415,3 +437,43 @@
...
+if hasattr(args, 'number_nutrition_plans'):
+    print("** Generating {0} nutrition plan(s) per user".format(args.number_nutrition_plans))
+
+    if args.add_to_user:
+        userlist = [User.objects.get(pk=args.add_to_user)]
+    else:
+        userlist = [i for i in User.objects.all()]
+
+    # Load all ingredients to a list
+    ingredientList = [i for i in Ingredient.objects.all()]
```
Comment (rolandgeider) [inline extras/dummy_generator/generator.py]: While this is a script that won't get used often, I'm not sure about loading all the ingredients at once. I'd perhaps do something like this (even if this is also not a perfect solution):

```python
ingredientList = [i for i in Ingredient.objects.order_by('?').all()[:100]]
```

File: extras/dummy_generator/generator.py [hunk 2]
```diff
@@ -415,3 +437,43 @@
...
+            nutrition_plan.user = user
+
+            nutrition_plan.save()
+
+            # Add meals to plan
+            order = 1
+            for j in range(0, total_meals):
+               meal = Meal(plan=nutrition_plan, order=order)
+               meal.save()
+               ingredient = ingredientList[random.randint(0, len(ingredientList))]
```
Comment (rolandgeider) [inline extras/dummy_generator/generator.py]: You can add another for loop here and add a random number of ingredients to the meal, perhaps something between 1 and 5?

**Comment by rolandgeider** [comment]
Thanks!

The PR looks good, I've added some comments, but that's really just nitpicking. Oh, don't forget to add yourself to the AUTHORS file :)

**Comment by rolandgeider** [comment]
I just saw you had added another commit, Github only sends notifications for new messages so I didn't catch it earlier. I've added another comment, but then I can merge this

File: extras/dummy_generator/generator.py [hunk 3]
```diff
@@ -415,3 +437,44 @@
...
+            # Add meals to plan
+            order = 1
+            for j in range(0, total_meals):
+                meal = Meal(plan=nutrition_plan, order=order)
+                meal.save()
+                for k in range(0, random.randint(1,5)):
+                    ingredient = ingredientList[random.randint(0, len(ingredientList)-1)]
+                    i_weight = IngredientWeightUnit(ingredient=ingredient, unit=WeightUnit.objects.all()[0], gram=random.randint(10, 250))
+                    i_weight.save()
+                    meal_item = MealItem(meal=meal, ingredient=ingredient, weight_unit=i_weight, order=order, amount=1)
```
Comment (rolandgeider) [inline extras/dummy_generator/generator.py]: I see you are generating new weight units, that's not necessary. You can just leave the weight_unit unset (or set it to None) then the system will use grams, for the amount you can just use a random integer. 

```python
meal_item = MealItem(meal=meal, [...] weight_unit=None, amount=random.randint(10, 250))
```

File: extras/dummy_generator/generator.py [hunk 4]
```diff
@@ -415,3 +437,44 @@
...
+
+            nutrition_plan.save()
+
+            # Add meals to plan
+            order = 1
+            for j in range(0, total_meals):
+                meal = Meal(plan=nutrition_plan, order=order)
+                meal.save()
+                for k in range(0, random.randint(1,5)):
+                    ingredient = ingredientList[random.randint(0, len(ingredientList)-1)]
```
Comment (rolandgeider) [inline extras/dummy_generator/generator.py]: Oh, and here you can also just do

```python 
ingredient = random.choice(ingredientList)
```
that's a bit cleaner

**Comment by cthare** [comment]
I actually meant to send a message with the previous commit, but I had something come up right after I submitted. I just made the other changes to remove the creation of a weight unit and cleaned up the ingredient line.

**Comment by rolandgeider** [comment]
Merged, thanks again.

**Pull request merged**